packages/venia-ui/lib/components/OrderHistoryPage/orderHistoryPage.js

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+
+import { useOrderHistoryPage } from '@magento/peregrine/lib/talons/OrderHistoryPage/useOrderHistoryPage';
+import { mergeClasses } from '@magento/venia-ui/lib/classify';
+
+import { Title } from '../Head';
+import defaultClasses from './orderHistoryPage.css';
+
+const PAGE_TITLE = `Order History`;
+const EMPTY_DATA_MESSAGE = `You don't have any orders yet.`;
+
+const OrderHistoryPage = props => {
+    const talonProps = useOrderHistoryPage();
+    const { data } = talonProps;
+
+    const classes = mergeClasses(defaultClasses, props.classes);
+
+    let pageContents;
+    if (!data) {
```

Comment:
Let's get ahead of `cache-and-network` and always use `loading && !data` to only render empty/loading states when we're loading and have nothing else to render.